File: semester_two/algoDat/public/code/vorlesung-01/cpp/MinSort.cpp
// Copyright 2017, University of Freiburg,
// Chair of Algorithms and Data Structures.
// Author: Hannah Bast <bast@cs.uni-freiburg.de>

#include <vector>
#include <utility>
#include <iostream>
#include "./MinSort.h"

// _____________________________________________________________________________
void MinSort::sort(std::vector<int>& v) {
  for (std::size_t i = 0; i + 1 < v.size(); i++) {
    // Find minimum in v[i..end].
    int minValue = v[i];
    int minIndex = i;
    for (std::size_t j = i + 1; j < v.size(); j++) {
      if (v[j] < minValue) {
        minValue = v[j];
        minIndex = j;
      }
    }
    // Swap minimum to front.
    std::swap(v[i], v[minIndex]);
  }
}

File: semester_two/algoDat/public/code/vorlesung-01/java/MinSort.java
// Copyright 2017, University of Freiburg,
// Chair of Algorithms and Data Structures.
// Author: Axel Lehmann <lehmann@cs.uni-freiburg.de>

/**
 * Class for MinSort algorithm.
 */
public class MinSort {

  /**
   * MinSort implementation.
   */
  public static void sort(int[] v) {
    for (int i = 0; i + 1 < v.length; i++) {
      // Find minimum in v[i, ..., n-1].
      int minValue = v[i];
      int minIndex = i;
      for (int j = i + 1; j < v.length; j++) {
        if (v[j] < minValue) {
          minValue = v[j];
          minIndex = j;
        }
      }
      // Swap minimum to front.
      v[minIndex] = v[i];
      v[i] = minValue;
    }
  }
}
